influx/query.go

```diff
@@ -10,6 +10,52 @@ import (
 	"github.com/influxdata/influxdb/influxql"
 )
 
+func TimeRangeAsEpochNano(expr influxql.Expr, now time.Time) (min, max int64, err error) {
+	tmin, tmax, err := influxql.TimeRange(expr)
+	if err != nil {
+		return 0, 0, err
+	}
+	if tmin.IsZero() {
+		min = time.Unix(0, influxql.MinTime).UnixNano()
+	} else {
+		min = tmin.UnixNano()
+	}
+	if tmax.IsZero() {
+		max = now.UnixNano()
+	} else {
+		max = tmax.UnixNano()
+	}
+	return
+}
+
+const WhereToken = "WHERE"
+
+func ParseTime(influxQL string, now time.Time) (time.Duration, error) {
+	start := strings.Index(strings.ToUpper(influxQL), WhereToken)
+	if start == -1 {
+		return 0, fmt.Errorf("not a relative duration")
+	}
+	start += len(WhereToken)
+	where := influxQL[start:]
+	cond, err := influxql.ParseExpr(where)
+	if err != nil {
+		return 0, err
+	}
+	nowVal := &influxql.NowValuer{
```

Comment:
Do you mean for this to be `NowValue`?

Reply:
@lukevmorris I imported `NowValuer` from InfluxDB here:
https://github.com/influxdata/influxql/blob/3921ab7858b6af8443fe0bef06b52608be72852c/ast.go#L5025

In a nutshell, NowValuer returns a particular time when `now()` is parsed in an influxql statement.